cmd/flags/flags.go

```diff
@@ -33,22 +34,55 @@ const (
 	CassandraStorageType = "cassandra"
 	// MemoryStorageType is the storage type flag denoting an in-memory store
 	MemoryStorageType = "memory"
+
+	runtimeMetricsFrequency        = "runtime-metrics-frequency"
+	spanStorageType                = "span-storage.type"
+	logLevel                       = "log-level"
+	dependencyStorageType          = "dependency-storage.type"
+	dependencyStorageDataFrequency = "dependency-storage.data-frequency"
 )
 
-// ErrUnsupportedStorageType is the error when dealing with an unsupported storage type
-var ErrUnsupportedStorageType = errors.New("Storage Type is not supported")
+// Flags holds flags configuration
+type Flags struct {
+	// Logging defines common setting for controlling logs
+	Logging logging
+	// RuntimeMetricsFrequency specifies how often Go runtime stats should be reported.
+	RuntimeMetricsFrequency time.Duration
+	// SpanStorage defines common settings for Span Storage.
+	SpanStorage spanStorage
+	// DependencyStorage defines common settings for Dependency Storage.
+	DependencyStorage dependencyStorage
+}
 
-// Logging defines common setting for controlling logs
-var Logging = logging{}
+// NewFlags creates Flags with initialized properties from viper
+func NewFlags() *Flags {
+	//make sure that viper is initialized correctly
+	CreateFlags().VisitAll(func(flag *flag.Flag) {
+		viper.SetDefault(flag.Name, flag.DefValue)
+	})
 
-// RuntimeMetricsFrequency specifies how often Go runtime stats should be reported.
-var RuntimeMetricsFrequency time.Duration
+	flags := new(Flags)
+	flags.RuntimeMetricsFrequency = viper.GetDuration(runtimeMetricsFrequency)
+	flags.SpanStorage.Type = viper.GetString(spanStorageType)
+	flags.Logging.Level = viper.GetString(logLevel)
+	flags.DependencyStorage.Type = viper.GetString(dependencyStorageType)
+	flags.DependencyStorage.DataFrequency = viper.GetDuration(dependencyStorageDataFrequency)
+	return flags
+}
 
-// SpanStorage defines common settings for Span Storage.
-var SpanStorage = spanStorage{}
+// CreateFlags creates flag set for Flags object
```

Comment:
`CreateFlags constructs a new minimal set of flags` (assuming this is what I think it is)